### ISSUE TITLE
Cannot copy a source database in an elastic pool to another server with no elastic pool

### ISSUE THREAD

**Issue by panesofglass** [Azure/azure-libraries-for-net#244]
I need to copy a SQL Database in an elastic pool on one server to another server without an elastic pool. However, the following code fails, noting that the elastic pool name of the source database does not exist on the target server.

``` fsharp
let copyDatabase =
    targetSqlServer.Databases.Define(target.Name)
        //.WithServiceObjective("P1") // Using 1.4.0
        //.WithStandardEdition(SqlDatabasePremiumServiceObjective.P1)  // Using 1.8.0
        .WithSourceDatabase(sourceDatabase)
        .WithMode(CreateMode.Copy)
        .Create()
```

As you can see above, I tried to work around this by using the `WithStandardEdition` (also tried the `WithServiceObjective`) method. However, these return an interface that is not compatible with then specifying a source database. [`New-AzureRmSqlDatabaseCopy`](https://docs.microsoft.com/en-us/powershell/module/azurerm.sql/new-azurermsqldatabasecopy?view=azurermps-5.6.0) allows specifying either a target service objective or elastic pool, so I expected the .NET Azure libraries would allow this as well.

This issue exists in versions 1.4.0 and 1.8.0.

**Comment by milismsft** [comment]
@panesofglass @@johnverbiest, while I was able to reproduce the issue, I believe this is more of a limitation form The Azure SQL management service side vs something within the SDK implementation. I will follow up with the Azure SQL team to find out why this restrictions are currently in place.

Meanwhile the only work around I can suggest (and it seems to work) is to do this in couple steps:
 - create/add a matching elastic pool (including name and edition) to the target SQL server
 - initiate the copy SQL Database command
 - update the target database by taking it out of the temporary elastic pool and specifying the desired service objective settings
 - delete the elastic pool on the target SQL server

**Comment by panesofglass** [comment]
I just ran a copy in PowerShell today in which I specified the ServiceObjective, so I know it is possible. Are you using a different set of REST APIs than the PowerShell commandlets?

**Comment by milismsft** [comment]
PowerShell and the .NET SDK are using the same REST API's. What is the exact command you run?
+ @btasdoven

**Comment by panesofglass** [comment]
Here are the commands I am currently using in PowerShell:

``` powershell
Write-Output "copying $SourceDbName to $CopyDbName on $CopyDbServerName"
New-AzureRmSqlDatabaseCopy -ResourceGroupName $DbResourceGroupName -ServerName $SourceDbServerName -DatabaseName $SourceDbName -CopyResourceGroupName $WebDbResourceGroupName -CopyServerName $CopyDbServerName -CopyDatabaseName $CopyDbName -ServiceObjectiveName 'P1'
Set-AzureRmSqlDatabase -ResourceGroupName $WebDbResourceGroupName -ServerName $CopyDbServerName -DatabaseName $CopyDbName -RequestedServiceObjectiveName 'S4'
```

I'm copying from a Premium-tier elastic pool, so I have to go to a `P1` first, but I don't have to copy to another elastic pool. I can specify the service objective.

**Comment by btasdoven** [comment]
@panesofglass, it looks like the response type incompatibility between `WithServiceObjective` and `WithSourceDatabase` is resolved with v1.7. Could you confirm if the below doesn't work for you even with v1.7 or v1.8?
```
let copyDatabase =
    targetSqlServer.Databases.Define(target.Name)
        .WithServiceObjective("P1")
        .WithSourceDatabase(sourceDatabase)
        .WithMode(CreateMode.Copy)
        .Create()
```

**Comment by panesofglass** [comment]
Yes, that is what I tried in 1.8, and it did not work. I did not try 1.7.

**Comment by milismsft** [comment]
Hi @panesofglass the Azure SQL management service is inconsistent in regards to when setting a service objective while copying a database is allowed and when is not. I was able to successfully copy a SQL database within a premium Elastic Pool to a different SQL server also setting the service objective if the Elastic Pool edition is Premium. Any other edition settings seems to trigger a service error when invoking the createOrUpdate(). Having this inconsistency makes it difficult to model it in the Fluent .NET SDK.
I can suggest an work around for now while I will try to see what can be done for v1.9:
`
var tempStage = sqlServer.Databases.Define("dbCopyName")
    .WithServiceObjective(ServiceObjectiveName.P1);

var dbCopyResult = ((Microsoft.Azure.Management.Sql.Fluent.SqlDatabaseOperations.Definition.IWithAllDifferentOptions)tempStage)
    .WithSourceDatabase(dbSource)
    .WithMode(CreateMode.Copy)
    .Create();
`

**Comment by panesofglass** [comment]
@milismsft thanks!

**Comment by hovsepm** [comment]
@johnverbiest , @panesofglass , @btasdoven We have release another set of fixes in SQL in release 1.9.1. Could you upgrade your nuget package references and let me know if the issue is resolved?

**Comment by roofiq** [comment]
Hello, is there any known compatibility issue between database in elastic pools and standalone server? I have issue when migrating one database in that way and it fails when updating.

Here is description:
[](https://github.com/MicrosoftDocs/azure-docs/issues/51784)
